esp/src/test-ui/tests/framework/pages/ActivitiesTest.java

```diff
@@ -80,19 +89,29 @@ private String getCurrentPage() {
         return "Invalid Page";
     }
 
-    private boolean testTabsForNavigationLinks(NavigationWebElement element) {
+    private boolean testTabsForNavigationLinks(NavigationWebElement element, StringBuilder msg) {
+        msg.delete(0, msg.length());
+        boolean retVal = true;
+                
         List<String> tabsList = URLConfig.tabsListMap.get(element.name());
+        int numOfElemets = tabsList.size();
+        double elementFound = 0.0;  // It is double for calculating ratio later.
 
         for (String tab : tabsList) {
             try {
                 WebElement webElement = Common.waitForElement(By.xpath("//a[text()='" + tab + "']"));
                 urlMap.get(element.name()).getUrlMappings().put(tab, new URLMapping(tab, webElement.getAttribute("href")));
+                elementFound += 1.0;
             } catch (TimeoutException ex) {
-                return false;
+                msg.append("'" + tab + "', ");
             }
         }
-
-        return true;
+        
+        if ( (elementFound / numOfElemets) < 0.5 ) {
```

Comment:
Need to protect against numElements == 0 (and return early?)

Reply:
Changed